### ISSUE TITLE
legend: Understand behavior (and difference) of "-C" and "-F+c"

### ISSUE THREAD

**Issue by yvonnefroehlich** [GenericMappingTools/gmt#8303]
_Related to_ https://github.com/GenericMappingTools/pygmt/pull/2985#issuecomment-1893771230.

For `legend`, there are two ways to add space between the legend entries and the box outline:
- The [**-C**](https://docs.generic-mapping-tools.org/dev/legend.html#c) flag, which allows specifying *dx* and *dy*.
- The **+c** modifier of the [ **-F**](https://docs.generic-mapping-tools.org/dev/legend.html#f) flag, which allows specifying all four gaps separately. However, for me, this only has an effect in case a width (**+w**) is given via **-D**.

Behavior (and difference) of these two ways appear difficult to understand.

```bash
gmt begin legend_gap_box_entries_GMT png

    gmt basemap -R-5/5/-5/5 -JX10c -Baf -B+t"no space modification"
    echo 0  1 | gmt plot -Sc0.3c -l"Circle"
    echo 0 -1 | gmt plot -Ss0.3c -l"Square"
    gmt legend -DJTR+jTR+o0.2c -F+gwhite+p1p

    gmt basemap -R-5/5/-5/5 -JX10c -Baf -B+t"+c to -F" -Xw+1c
    echo 0  1 | gmt plot -Sc0.3c -l"Circle"
    echo 0 -1 | gmt plot -Ss0.3c -l"Square"
    gmt legend -DJTR+jTR+o0.2c -F+p1p,black+c0.1c/0.4c/1c/0.5c

    gmt basemap -R-5/5/-5/5 -JX10c -Baf -B+t"+c to -F and \053w to -D" -Xw+1c
    echo 0  1 | gmt plot -Sc0.3c -l"Circle"
    echo 0 -1 | gmt plot -Ss0.3c -l"Square"
    gmt legend -DJTR+jTR+o0.2c+w2c -F+p1p,black+c0.1c/0.4c/1c/0.5c

    gmt basemap -R-5/5/-5/5 -JX10c -Baf -B+t"-C flag of legend" -Xw+1c
    echo 0  1 | gmt plot -Sc0.3c -l"Circle"
    echo 0 -1 | gmt plot -Ss0.3c -l"Square"
    gmt legend -DJTR+jTR+o0.2c -C0.5c/1c -F+gwhite+p1p

gmt end show
````
_Output figure_:
![legend_gap_box_entries_GMT](https://github.com/GenericMappingTools/gmt/assets/94163266/6ccc930b-55fa-4cef-b3c4-bd5448fda895)

**Comment by PaulWessel** [comment]
Good question. Looking at the code we determine the size of the legend box in two different ways:

1. Via **-D+w**_width_[/_height_] which specifically determines the dimensions.  Here, _height_ is set to _width_ if not given (square legend box).
2. With no explicit dimensions given then we determine the required dimensions using _PostScript_ operators based on the contents of the label and create the PostScript parameters _PSL_legend_label_width_ and _PSL_legend_box_height_. These are then used internally to draw the box.

Now, the **-C**_dx_[/_dy_] option can be used to extend the box in two ways.  I will update the doc syntax to show that _dx_ is optional and set to _dx_ if not given, like in most places in GMT.

Then we have the **-F+c**_clearance_ modifier which seems to overlap with **-C** here. Certainly, makes little sense to me to allow both at the same time.  I think the intent for **+c** was to allow more flexibility in even adding specific clearance that differ on all four sides.  Perhaps adding space where a color bar might fit.

I will test the behaviour of not allowing both **-C** and **-F+c** in the same command. I think that is the only sensible solution since **-C** cannot fully replicate **+c**'s flexibility anyway.

**Comment by Esteban82** [comment]
This was fixed by #8305, right?

**Comment by PaulWessel** [comment]
Yes
